onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -172,10 +172,41 @@ Status Transpose<float>::Compute(OpKernelContext* ctx) const {
   return Status::OK();
 }
 
-ONNX_CPU_OPERATOR_KERNEL(
+template <>
+Status Transpose<int64_t>::Compute(OpKernelContext* ctx) const {
+  // Get input and output:
+  const Tensor* input_tensor_ptr = ctx->Input<Tensor>(0);
+  ORT_ENFORCE(input_tensor_ptr != nullptr);
+  const Tensor& X = *input_tensor_ptr;
+  const TensorShape& input_shape = X.Shape();
+  const std::vector<int64_t>& input_dims = input_shape.GetDims();
+  size_t rank = input_dims.size();
+
+  std::vector<int64_t> output_dims(rank);
+  const std::vector<int64_t>* p_perm;
+  std::vector<int64_t> default_perm(rank);
+  ComputeOutputShape(X, output_dims, default_perm, p_perm);
+
+  TensorShape output_shape{output_dims};
+  Tensor& Y = *ctx->Output(0, output_shape);
+
+  DoTypedTranspose<int64_t>(*p_perm, X, Y);
+
+  return Status::OK();
+}
+
+ONNX_CPU_OPERATOR_TYPED_KERNEL(
     Transpose,
     1,
+    float,
     KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
     Transpose<float>);
 
+ONNX_CPU_OPERATOR_TYPED_KERNEL(
+    Transpose,
+    1,
+    int64_t,
```

Comment:
>int64_t [](start = 4, length = 7)

Ideally, all tensor ops should not register typed kernel to avoid bloating binary size with templates. As an example, Reshape have been changed to a generic kernel. Would be nice if this PR can do that to Transpose.